mastering/editable_packages.rst

```diff
@@ -0,0 +1,131 @@
+
+and we need a file to tell Conan what is the layout of your local project in order to provide the
+proper paths to the build system. There are several ways to achieve this objective:
+
+ * A file next to the *conanfile.py* with the name *.conan_layout*:
+
+   .. code-block:: ini
+
+       [includedirs]
+       src/core/include
+       src/cmp_a/include
+
+       [libdirs]
+       build/{settings.build_type}/{options.shared}
+
+       [bindirs]
+       build/{settings.build_type}/{options.shared}
+
+ * A file inside the Conan cache in the path *layouts/default*. As this file can contain information
+   for several packages, each section must declare which packages it applies to, this is achieved
+   using a *namespace* with the name of the package or the wildcard ``*``:
+
+   .. code-block:: ini
+
+       [cool:includedirs]
+       src/core/include
+       src/cmp_a/include
+
+       [*:libdirs]
+       build/{settings.build_type}/{options.shared}
+
+       [*:bindirs]
+       build/{settings.build_type}/{options.shared}
+
+
+   This file can be very handy inside a company where all the packages have the same layout.
+
+As you can see, you can use inside these files some **placeholders** that will be substituted with
+the values of the ``settings`` and the ``options`` of the package.
+
+About the precedence if both files are present, Conan will use the one inside the repository if
+present, then the one in the Conan cache, and if none is available, Conan will fallback to the
+directories defined in the ``package_info`` method of the recipe itself but related to the path
+where the editable package has been linked to.
+
+
+Using a package in editable mode
+--------------------------------
+
+Once a package is in editable mode it is used **system wide** by Conan (by every Conan client that
+uses the same cache), no changes are required in the consumers. Every :command:`conan install`
+command that requires our editable ``cool/version@user/dev`` package will use the paths to
+the local directory and the changes made to this project will be taken into account by the
+packages using its headers or linking against it.
+
+
+Revert the editable mode
+------------------------
+
+In order to revert the editable mode, only the link has to be removed, there is command for that:
+
+.. code-block:: bash
+
+    $ conan link --remove cool/version@user/dev
+
+It will remove the link (the local directory won't be affected) and all the packages using this
+requirement will use the one in the cache again.
+
+
+.. warning::
+
+   Packages that are built using an editable package in its graph upstreams can generate binaries
+   and packages incompatible with the released version of the editable package. Avoid uploading
+   these packages without re-creating them with the in-cache version of all the libraries.
+
+
+.. note::
+
+   This is an experimental feature and it is in its early stage, it can suffer changes on its
```

Comment:
Place this warning at the top of the section like the other ones to warn users from the beginning